aggregation-worker/src/main/java/au/org/emii/aggregationworker/AggregationRunner.java

```diff
@@ -95,40 +127,31 @@ public void run(String... args) {
 
             Options options = new Options();
 
-            options.addOption("b", true, "restrict to bounding box specified by left lower/right upper coordinates e.g. -b 120,-32,130,-29");
-            options.addOption("z", true, "restrict data to specified z index range e.g. -z 2,4");
-            options.addOption("t", true, "restrict data to specified date/time range in ISO 8601 format e.g. -t 2017-01-12T21:58:02Z,2017-01-12T22:58:02Z");
-            // reexamine how config would be passed to program - as an argument perhaps?
-            options.addOption("c", true, "aggregation overrides to be applied - xml");
+            options.addOption("l", true, "The layer name");
+            options.addOption("s", true, "Subset parameters");
+            options.addOption("m", true, "The requested output mime type");
+            options.addOption("e", true, "Callback email address");
 
             CommandLineParser parser = new DefaultParser();
             CommandLine commandLine = parser.parse(options, args);
 
-            //  TODO:  Parameters are currently passed as command line arguments & are positional.
-            //  TODO:  need to revisit to accept named parameters passed in any order.
-            //  TODO:  Some scheme like passive them as a <name>=<value> pair for each one might be workable
-            //  TODO:  (as long as the value doesn't include spaces)
-            //
-            //  Currently expects the following params (IN THIS ORDER) -
-            //        - layer
-            //        - subset
-            //        - result
-            String layer = commandLine.getArgs()[0];
-            String subset = commandLine.getArgs()[1];
-            String resultMime = commandLine.getArgs()[3];
-            email = commandLine.getArgs()[2];
-
-            if (email != null && email.startsWith("email.to=")) {
-                email = email.substring(9); // To support current request format
-            }
-
-            logger.info("email:" + email);
+            String layer = commandLine.getOptionValue('l');
+            String subset = commandLine.getOptionValue('s');
+            String resultMime = commandLine.getOptionValue('m');
+            email = commandLine.getOptionValue('e');
 
             SubsetParameters subsetParams = new SubsetParameters(subset);
 
-            logger.info("Layer name        = " + layer);
-            logger.info("Subset parameters = " + subset);
-            logger.info("Request MIME type = " + resultMime);
+            if(email != null) {
+                email = email.substring(email.indexOf("=") + 1);
```

Comment:
Would ```email.split('=')[1]``` be cleaner?

Reply:
Depends who is looking at it I guess  ;)